tests/TaskLineRenderer.test.ts

```diff
@@ -279,6 +279,23 @@ describe('task line rendering', () => {
         );
     });
 
+    it.each([
+        ['a task with a tag at the end #todo', 'a task with a tag at the end'],
+        ['a task with a tag #tag in the middle', 'a task with a tag in the middle'],
+        ['#now a task with a tag in the beginning', 'a task with a tag in the beginning'],
+        ['#important #s a task with #today #TODO a lot #never of tags #now #PERFORMANCE', 'a task with a lot of tags'],
+    ])(
+        'renders without tags task with description "%s"',
+        async (taskDescription: string, renderedDescription: string) => {
+            await testLayoutOptions(
+                `- [ ] ${taskDescription} ⏫ 📅 2022-07-02 ⏳ 2022-07-03 🛫 2022-07-04 ➕ 2022-07-05 🔁 every day`,
+                { hideTags: true },
+                renderedDescription,
+                [' ⏫', ' 🔁 every day', ' ➕ 2022-07-05', ' 🛫 2022-07-04', ' ⏳ 2022-07-03', ' 📅 2022-07-02'],
+            );
+        },
```

Comment:
I noticed that there weren't any tests for nested tags, so I tried it out manually.

Filename: `asdfasdfads.md`

~~~
- [ ] #task Review [feat: \`hide tags\` instruction by ilandikov · Pull Request #2076 · obsidian-tasks-group/obsidian-tasks · GitHub](https://github.com/obsidian-tasks-group/obsidian-tasks/pull/2076)
- [ ] #task #a/b/c stuff

```tasks
group by tags
path includes asdfasdfads
hide tags
```
~~~

What is rendered - note the line containing `/b/c stuff`:

![image](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/4840096/c3122ca0-5440-47e9-8654-f582007f0f1f)

Reply:
Another example is hyphens in tags:

~~~
- [ ] #task using #a-tag-with-hyphens

```tasks
group by tags
path includes asdfasdfads
hide tags
```
~~~

Gives

![image](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/4840096/52b69dd1-8a9d-40b6-ba2a-5e0406f38a84)

Reply:
Many thanks for the extensive examples, I added them =)

Reply:
Shall be fixed now =)